spec/spec_helper_acceptance.rb

```diff
@@ -56,25 +56,35 @@
             on(host, 'apt-get install -y zlib1g-dev patch g++')
           end
 
-          gem_command = if is_windows?(host)
-                          'cmd.exe /c cmd.exe /c "C:\Program Files\Puppet Labs\puppet\sys\ruby\bin\gem.bat"'
-                        else
-                          host.file_exist?("#{host['privatebindir']}/gem") ? "#{host['privatebindir']}/gem" : "#{host['puppetbindir']}/gem"
-                        end
+          gems = [
+              # Unf has a separate gem for it's native extensions, unf_ext. Unf_ext has windows packages with the precompiled
+              # libraries. Because of this setup just installing the top level azure gems doesn't always seem to do the
+              # right thing on Windows
+            [ 'unf' ],
+            [ 'hocon', 'retries' ],
+            # Azure gems require pinning because they are still under heavy development and change their API frequently
+            [ 'azure_mgmt_compute', '--version=~> 0.3.0' ],
+            [ 'azure_mgmt_network', '--version=~> 0.3.0' ],
+            [ 'azure_mgmt_resources', '--version=~> 0.3.0' ],
+            [ 'azure_mgmt_storage', '--version=~> 0.3.0' ],
+            [ 'azure', '--version=~> 0.7.0' ],
+          ]
+
+          additional_gem_opts = ["--no-ri", "--no-rdoc"]
+
           if is_windows?(host)
-            # Unf has a separate gem for it's native extensions, unf_ext. Unf_ext has windows packages with the precompiled
-            # libraries. Because of this setup just installing the top level azure gems doesn't always seem to do the
-            # right thing on Windows
-            on(host, "#{gem_command} install unf")
+            # shield the quoting from beaker's autoquoting to correctly quote the space
+            # also avoid the gem.bat, as that cannot pass arguments with spaces and other special characters through
+            windows_cmd = [ '/cygdrive/c/Program Files/Puppet Labs/Puppet/sys/ruby/bin/ruby.exe', 'C:\Program Files\Puppet Labs\Puppet\sys\ruby\bin\gem' , 'install' ]
```

Comment:
I don't like it a bit, but we only test on very specific versions of the
aio agent, do not support other rubies, and - most damning - I didn't find
any other way to invoke gem with args that contains spaces through beaker.
On Jul 13, 2016 16:56, "Glenn" <notifications@github.com> wrote:

> In spec/spec_helper_acceptance.rb
> <https://github.com/puppetlabs/puppetlabs-azure/pull/95#discussion_r70654060>
> :
>
> >            if is_windows?(host)
> > -            # Unf has a separate gem for it's native extensions, unf_ext. Unf_ext has windows packages with the precompiled
> > -            # libraries. Because of this setup just installing the top level azure gems doesn't always seem to do the
> > -            # right thing on Windows
> > -            on(host, "#{gem_command} install unf")
> > +            # shield the quoting from beaker's autoquoting to correctly quote the space
> > +            # also avoid the gem.bat, as that cannot pass arguments with spaces and other special characters through
> > +            windows_cmd = [ '/cygdrive/c/Program Files/Puppet Labs/Puppet/sys/ruby/bin/ruby.exe', 'C:\Program Files\Puppet Labs\Puppet\sys\ruby\bin\gem' , 'install' ]
>
> Is it a little dangerous to hard code to this path?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-azure/pull/95/files/d7824f96c0da0c91842a0c6255caeddbc59c9649#r70654060>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABsQDPireFYg1PsYpfVFgp79FCK1V4Yks5qVQqegaJpZM4JLSVv>
> .
>